src/Core/App.php

```diff
@@ -172,38 +177,63 @@ protected static function _classExistsInBase(string $name, string $namespace): b
      *
      * Will return the value of `App.paths.plugins` config.
      *
-     * ```
-     * App::path('Model/Table');
-     * ```
+     * Deprecated: 4.0 App::path() is deprecated for class path (inside src/ directory).
+     *   Use \Cake\Core\App::classPath() instead or directly the method on \Cake\Core\Plugin class.
      *
-     * Will return the path for tables `src/Model/Table`.
-     *
-     * @param string $type type of path
-     * @param string|null $plugin This argument is deprecated.
-     *   Use \Cake\Core\Plugin::classPath()/templatePath() instead for plugin paths.
-     * @return array
+     * @param string $type Type of path
+     * @param string|null $plugin Plugin name
+     * @return string[]
      * @link https://book.cakephp.org/4/en/core-libraries/app.html#finding-paths-to-namespaces
      */
     public static function path(string $type, ?string $plugin = null): array
     {
-        if (empty($plugin)) {
-            if ($type[0] === strtolower($type[0])) {
-                return (array)Configure::read('App.paths.' . $type);
-            }
+        if ($plugin === null && $type[0] === strtolower($type[0])) {
+            return (array)Configure::read('App.paths.' . $type);
+        }
 
-            return [APP . $type . DIRECTORY_SEPARATOR];
+        if ($type === 'templates') {
+            return [Plugin::templatePath($plugin)];
         }
 
         deprecationWarning(
-            'Using App::path() with 2nd argument $plugin is deprecated.'
-            . ' Use \Cake\Core\Plugin::classPath()/templatePath() instead.'
+            'App::path() is deprecated for class path.'
+            . ' Use \Cake\Core\App::classPath() or \Cake\Core\Plugin::classPath() instead.'
         );
 
-        if ($type === 'templates') {
-            return [Plugin::templatePath($plugin)];
+        return static::classPath($type, $plugin);
+    }
+
+    /**
+     * Gets the path to a class type in the application or a plugin.
+     *
+     * Example:
+     *
+     * ```
+     * App::classPath('Model/Table');
+     * ```
+     *
+     * Will return the path for tables `src/Model/Table`.
+     *
+     * ```
+     * App::classPath('My/Plugin.Model/Table');
```

Comment:
The argument still doesn't seem right to me. Shouldn't it be `App::classPath('Model/Table', 'My/Plugin')`?

Reply:
I guess my brain was too much into Router::path()^^